applications/portfolioDrafts/portfolio/createPortfolioStep.ts

```diff
@@ -33,25 +53,34 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
   const command = new UpdateCommand({
     TableName: TABLE_NAME,
     Key: {
-      id: portfolioId,
+      id: portfolioDraftId,
     },
-    UpdateExpression: "set #portfolioVariable = :x",
+    UpdateExpression: "set #portfolioVariable = :x, updated_at = :y",
     ExpressionAttributeNames: {
       "#portfolioVariable": "portfolio_step",
     },
     ExpressionAttributeValues: {
       ":x": portfolioStep,
+      ":y": now,
     },
+    ConditionExpression: "attribute_exists(created_at)",
+    ReturnValues: "ALL_NEW",
   });
 
   try {
     await client.send(command);
-  } catch (err) {
-    console.log("Database error: " + err);
+    return new ApiSuccessResponse<PortfolioStep>(portfolioStep, SuccessStatusCode.CREATED);
+  } catch (error) {
+    console.log("Database error: " + error.name);
+    if (error.name === "ConditionalCheckFailedException") {
+      return new ErrorResponse(
+        { code: ErrorCodes.INVALID_INPUT, message: "Portfolio Draft with the given ID does not exist" },
+        ErrorStatusCode.BAD_REQUEST
```

Comment:
```suggestion
        ErrorStatusCode.NOT_FOUND
```

Per the spec, this is a 404